### ISSUE TITLE
mmio parse error for .mtx files with headers starting with spaces

### ISSUE THREAD

**Issue by istispl** [scipy/scipy#9426]
My issue is about mmio not parsing .mtx files if the header lines start with spaces.
For example, loading the attached .mtx file fails.

The header parser could be relaxed by stripping the spaces when reading header lines:
```
    @classmethod
    def info(self, source):
          ....
            # read and validate header line
            line = stream.readline().strip()
             ....
            # skip comments
            while line.startswith(b'%'):
                line = stream.readline().strip()
```

### Reproducing code example:

```
import scipy.io

print(scipy.io.mminfo('input_matrix_8.mtx'))
```

### Error message:
```
Header line not of length 3
Traceback (most recent call last):
  File "C:\anaconda2\lib\site-packages\scipy\io\mmio.py", line 258, in info
    raise ValueError("Header line not of length 3: " + line)
```

### Scipy/Numpy/Python version information:
```
('1.1.0', '1.14.3', sys.version_info(major=2, minor=7, micro=15, releaselevel='final', serial=0))
```

[input_matrix_8.zip](https://github.com/scipy/scipy/files/2534073/input_matrix_8.zip)

**Comment by WarrenWeckesser** [comment]
Where did you get that file?  How was it generated?

In Section 2 of the documentation at https://math.nist.gov/MatrixMarket/reports/MMformat.ps, in the description of the header, it says

> The first 15 characters must be `%%MatrixMarket` followed by at least one blank.

It doesn't say that blanks *in front of* `%%MatrixMarket` are allowed.  I'm not an expert in this format, but my first impression is that your file does not comply with the specification.  Maybe other folks more familiar with this format can confirm this.

**Comment by istispl** [comment]
Indeed, the attached file doesn't strictly obey the spec - and it is debatable to be tolerant on the parsing side. But since the fix is trivial, I considered reporting it, more as 'enhancement request' then than a defect.

**Comment by sethtroisi** [comment]
@WarrenWeckesser Do you still consider this an issue if we are correctly adhere to the spec?

**Comment by WarrenWeckesser** [comment]
The specification is pretty clear about how the file must start, so my preference is to push back on whoever created the file, and get them to follow the specification.  I'm not going to close this yet, because it looks like #12177 didn't completely fix the bug  in the `raise` statement.  I'll add details over there.

**Comment by s-m-h** [comment]
While the spec is clear about not allowing leading white space it appears that MUMPS in WRITE_ELEMENT is writing files with leading spaces. mmread in MATLAB happily ignores a leading white space. it would be nice if python would be as forgiving to matrices coming from MUMPS matrix export

**Comment by AtsushiSakai** [comment]
I confirmed #18289 fixed this issue from scipy 1.11.0
So, let's close this issue

# SciPy 1.10.0
```
In [4]: scipy.io.mminfo("/Users/atsushi_sakai/Desktop/input_matrix_8.mtx")
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[4], line 1
----> 1 scipy.io.mminfo("/Users/atsushi_sakai/Desktop/input_matrix_8.mtx")

File ~/anaconda3/lib/python3.11/site-packages/scipy/io/_mmio.py:79, in mminfo(source)
     31 def mminfo(source):
     32     """
     33     Return size and storage parameters from Matrix Market file-like 'source'.
     34
   (...)
     77     (5, 5, 7, 'coordinate', 'real', 'general')
     78     """
---> 79     return MMFile.info(source)

File ~/anaconda3/lib/python3.11/site-packages/scipy/io/_mmio.py:405, in MMFile.info(self, source)
    403 else:
    404     if not len(split_line) == 3:
--> 405         raise ValueError("Header line not of length 3: " +
    406                          line.decode('ascii'))
    407     rows, cols, entries = map(int, split_line)
    409 return (rows, cols, entries, format, field.lower(),
    410         symmetry.lower())

ValueError: Header line not of length 3:  %%MatrixMarket matrix coordinate complex general
```

# SciPy 1.11.0
```
In [5]: scipy.io.mminfo("/Users/atsushi_sakai/Desktop/input_matrix_8.mtx")
Out[5]: (8, 8, 64, 'coordinate', 'complex', 'general')
```

# SciPy 1.14.1
```
In [5]: scipy.io.mminfo("/Users/atsushi_sakai/Desktop/input_matrix_8.mtx")
Out[5]: (8, 8, 64, 'coordinate', 'complex', 'general')
```
